packages/authentication/src/services/authentication-module.ts

```diff
@@ -37,6 +43,38 @@ export default class AuthenticationModuleService<
   TAuthProvider extends AuthProvider = AuthProvider
 > implements AuthenticationTypes.IAuthenticationModuleService
 {
+  __joinerConfig(): ModuleJoinerConfig {
+    return joinerConfig
+  }
+
+  __hooks = {
+    onApplicationStart: async () => {
+      const providersToLoad = this.__container__["auth_providers"]
+
+      const providers = await this.authProviderService_.list({
+        provider: providersToLoad.map((p) => p.provider),
+      })
+
+      const loadedProvidersMap = new Map(providers.map((p) => [p.provider, p]))
+
+      const providersToCreate: ServiceTypes.CreateAuthProviderDTO[] = []
+
+      for (const provider of providersToLoad) {
+        if (loadedProvidersMap.has(provider.provider)) {
+          continue
+        }
+
+        providersToCreate.push({
+          provider: provider.provider,
+          name: provider.displayName,
+        })
+      }
+
+      await this.authProviderService_.create(providersToCreate)
```

Comment:
suggestion: Could you move that to a private method and call it here just to not surcharge the hook readability